tests/test_main.py

```diff
@@ -206,7 +216,17 @@ def run(self, parsed_args):
                 """
             ),
         ],
-        ["type: charm", None],
+        [
```

Comment:
The old version test should still work, right?

Reply:
No, "type: charm" only, without metadata will now fail pydantic model.

Reply:
Ahh, okay. Makes sense.

I'm trying to determine whether there are any use cases here that we're breaking that we care not to break. For example, if a user wants to run a store command on a custom store they might create a `charmcraft.yaml` file that has just `type: charm` and a `charmhub` section. I'm not sure we care about backwards compatibility with that case though.

Reply:
I don't think that a charm without a name is allowed in any case.

Reply:
It's not — the idea is that this would be a file that just contains charmcraft configuration for use with commands that don't require a charm. However, in the last day I haven't come up with a reason to support this use case, so I'm going to let it rest.